### DISCUSSION TITLE
How to rotate the iTwin model using ModelDisplayTransform and ModelDisplayTransformProvider

### DISCUSSION THREAD

**Discussion by ilyasbelfar** [iTwin/itwinjs-core#7039] [Q&A] [answered]
Hello people,
I read in the iTwin.js documentation that the ModelDisplayTransform can apply transformations at model's display time to the model loaded inside the iTwin viewer viewport (screenshots attached). How can ModelDisplayTransform and ModelDisplayTransformProvider be used to rotate the iTwin model along the x, y, and z axes at the model's display time? How should this be implemented?

Any guidance or examples would be greatly appreciated!

Here is the links to the iTwin.js docs:

- https://www.itwinjs.org/reference/core-frontend/views/modeldisplaytransform/
- https://itwinjs.org/reference/core-frontend/views/viewstate/modeldisplaytransformprovider/

![Capture](https://github.com/user-attachments/assets/d034faa4-ec65-4209-add9-db1492302802)

![Capture_2](https://github.com/user-attachments/assets/3ba2e7bf-85ca-47d5-b930-59f5aac24644)

**Comment by pmconne** [comment, marked as answer]
Take a look at [this simple tool](https://github.com/iTwin/itwinjs-core/blob/64a08a35daf4b7f44b5730faebfee743c94656b9/test-apps/display-test-app/src/frontend/DisplayTransform.ts#L83) that applies a transform to the set of models displayed in a viewport using `ModelDisplayTransformProvider`. You should be able to extract the bits you need.
